src/data/decisionDesk/services.ts

```diff
@@ -0,0 +1,263 @@
+            scope.setFingerprint([name])
+            scope.setTags({ domain: 'fetchBearerToken' })
+            scope.setTag('attemptCount', attemptCount)
+            scope.setTransactionName(name)
+          },
+        },
+      ),
+    {
+      retries: 1,
+      minTimeout: 4000,
+    },
+  )
+
+  logger.debug(`fetchBearerToken returned with status ${response.status}`)
+
+  const json = (await response.json()) as GetBearerTokenResponse | { errors: any[] }
+
+  if ('errors' in json) {
+    throw new Error(`fetchBearerToken threw with ${JSON.stringify(json.errors)}`)
+  }
+
+  return json
+}
+
+export async function getBearerToken() {
+  logger.debug('getBearerToken called')
+
+  const hasCachedBearerToken = await redis.get<GetBearerTokenResponse>(DECISION_DESK_BEARER_TOKEN)
+
+  if (hasCachedBearerToken) {
+    logger.debug('getBearerToken found cached token')
+    return hasCachedBearerToken.access_token
+  }
+
+  logger.debug('getBearerToken did not find cached token')
+
+  const bearerToken = await fetchBearerToken()
+
+  await redis.set(DECISION_DESK_BEARER_TOKEN, bearerToken, {
+    ex: bearerToken.expires_in,
+  })
+
+  logger.debug('getBearerToken set new token in cache')
+
+  return bearerToken.access_token
+}
+
+export async function fetchRacesData(params?: GetRacesParams) {
+  logger.debug('fetchRacesData called')
+
+  const endpointURL = new URL(`${API_ENDPOINT}/races`)
+
+  if (params) {
+    logger.debug('fetchRacesData received params', params)
+
+    const validationResult = GetRacesParamsSchema.safeParse(params)
+
+    if (!validationResult.success) {
+      throw new Error(
+        `fetchRacesData received invalid params ${JSON.stringify(validationResult.error)}`,
+      )
+    }
+
+    const URLParams = new URLSearchParams(
+      Object.entries(params).reduce(
+        (acc, [key, value]) => {
+          if (value !== undefined) {
+            acc[key] = value
+          }
+          return acc
+        },
+        {} as Record<string, any>,
+      ),
+    )
+
+    endpointURL.search = URLParams.toString()
+  }
+
+  const bearerToken = await getBearerToken()
+
+  if (!bearerToken) {
+    throw new Error('Bearer key not found')
+  }
+
+  const response = await pRetry(
+    attemptCount =>
+      fetchReq(
+        endpointURL.href,
+        {
+          headers: {
+            Authorization: `Bearer ${bearerToken}`,
+          },
+        },
+        {
+          withScope: scope => {
+            const name = `fetchRacesData attempt #${attemptCount}`
+            scope.setFingerprint([name])
+            scope.setTags({ domain: 'fetchRacesData' })
+            scope.setTag('attemptCount', attemptCount)
+            scope.setTransactionName(name)
+          },
+        },
+      ),
+    {
+      retries: 1,
+      minTimeout: 4000,
+    },
+  )
+
+  logger.debug(`fetchRacesData returned with status ${response.status}`)
+
+  const json = (await response.json()) as { data: GetRacesResponse } | { errors: any[] }
```

Comment:
The type cast is not matching the api return. It should be `GetRacesResponse` instead of `{ data: GetRacesResponse }` 
```suggestion
  const json = (await response.json()) as GetRacesResponse | { errors: any[] }
```